### PR TITLE
[java] Add static method type resolution

### PR DESCRIPTION
**Pull request by WinterGrascph**

### CODE REVIEW

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java [hunk 1]
```diff
@@ -409,21 +411,19 @@ public Object visit(ASTName node, Object data) {
 
         JavaTypeDefinition previousType;
 
-        // TODO: static method invocation
-        if (dotSplitImage.length == 1 && astArguments != null) { // method
+        if (node.getType() != null) { // static field or method
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java]: I'm lost here... who is setting this on a static access? I don't see it being set in this method up to this point, and `ASTName` has no children....

**Comment by jsotuyod** [comment]
I assume this has issues with method overloads (ie: `Assert.assertEquals`), as the non-static code does, right? We should add a TODO / test cases for it.

Same applies to shadowing between a local static method + a statically imported one (star import or not).

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java (hunk 1 above)
Comment (WinterGrascph) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java]: It is set in searchNodeNameForClass. I'll add a comment to clarify this.

**Comment by WinterGrascph** [comment]
This PR just enables potentially applicable static methods to be found in static imports and then the code handling the overload resolution is the same as with non-static methods. So the same problems persist with shadowing.

**Comment by jsotuyod** [comment]
Merged!

**Pull request merged**